app/controllers/divisions_controller.rb

```diff
@@ -32,7 +32,13 @@ def index
       elsif params[:date] =~/^\d{4}-\d{2}$/
         @month = params[:date]
       else
-        @date = params[:date]
+        begin
+          @date = Date.parse(params[:date]) if params[:date]
+        rescue ArgumentError => e
+          invalid_date = params[:date]
+          Rails.logger.info "Invalid date #{invalid_date}"
```

Comment:
I am know sure about this line, for two reasons:

1) Not sure if this is an error worth logging

2) It definitely needs better wording. 

Let me know if you want me to change this.